python/dgl/graphbolt/impl/torch_based_feature_store.py

```diff
@@ -99,9 +110,9 @@ def update(self, value: torch.Tensor, ids: torch.Tensor = None):
             )
             self._tensor[:] = value
         else:
-            assert ids.shape[0] == value.shape[0], (
-                f"ids and value must have the same length, "
-                f"but got {ids.shape[0]} and {value.shape[0]}."
+            assert self._tensor[ids].shape == value.shape, (
```

Comment:
we do the sanity check in `GPUCachedFeature` or `TorchBasedFeature`, is it possible to do such checks in a more unified way?

Reply:
> we do the sanity check in `GPUCachedFeature` or `TorchBasedFeature`, is it possible to do such checks in a more unified way?

This change will be withdrawn, so that the checks will be in a unified way. I suggest that this check can be removed in `GPUCachedFeature` for it will call the `update API` in `TorchBasedFeature`? Also, a check with feature size will be added once the modification of feature class have been done.